packages/visx-xychart/src/components/XYChart.tsx

```diff
@@ -168,7 +167,23 @@ export default function XYChart<
     );
   }
 
-  return width > 0 && height > 0 ? (
+  return (
+    <XYChartInternal captureEvents={captureEvents} accessibilityLabel={accessibilityLabel}>
+      {children}
+    </XYChartInternal>
+  );
+}
+
+function XYChartInternal(props: {
+  captureEvents: boolean;
+  accessibilityLabel: string;
+  children: React.ReactNode;
+}) {
+  const { captureEvents, accessibilityLabel, children } = props;
+  const eventEmitters = useEventEmitters({ source: XYCHART_EVENT_SOURCE });
+  const { width, height, margin } = useContext(DataContext);
```

Comment:
Created an internal component so I could read `width`, `height`, and `margin` off of the `DataContext`. Doing that in one component felt a little convoluted? 

This also moved `useEventEmitters`. Is there an issue if `useEventEmitters` is called after `useEventHandlers`?

Reply:
I can't check, but I _believe_ this extra component is what is causing happo to fail?